libtransact/src/execution/executor/mod.rs

```diff
@@ -100,13 +90,62 @@ impl Executor {
     }
 }
 
-impl SubSchedulerHandler for Executor {
+/// The interface for submitting execution tasks to the Executor.
+#[derive(Clone)]
+pub struct ExecutionTaskSubmitter {
+    sender: Sender<ExecutorCommand>,
+    readers: Arc<Mutex<HashMap<usize, ExecutionTaskReader>>>,
+}
+
+impl ExecutionTaskSubmitter {
+    /// Submit an Iterator of Execution tasks and a completion notifier to the Executor for
+    /// processing.  The iterator provided will be consumed until either it is exhausted or the
+    /// Executor is shutdown.
+    ///
+    /// # Errors
+    ///
+    /// Returns an `ExecutorError`if the tasks cannot be successfully sent to the Executor, either
+    /// due to the execution being shutdown .
```

Comment:
Missing a space in the first line, and looks like some missing info in the second line.

Reply:
Fixed